src/sync/update_tx.js

```diff
@@ -97,6 +98,60 @@ async function onSuccessfulTx(tx, db) {
   let txid;
 
   switch (tx.type) {
+    // Approve was accepted. Sending createEvent.
+    case 'APPROVECREATEEVENT': {
+      try {
+        const createTopicTx = await eventFactory.createTopic({
+          oracleAddress: tx.resultSetterAddress,
+          eventName: tx.name,
+          resultNames: tx.options,
+          bettingStartTime: tx.bettingStartTime,
+          bettingEndTime: tx.bettingEndTime,
+          resultSettingStartTime: tx.resultSettingStartTime,
+          resultSettingEndTime: tx.resultSettingEndTime,
+          senderAddress: tx.senderAddress,
+        });
+        txid = createTopicTx.txid;
+      } catch (err) {
+        logger.error(`Error calling EventFactory.createTopic: ${err.message}`);
+        throw err;
+      }
+
+      // Remove old Topic and insert new one with correct id
+      const topic = await db.Topics.findOne({ txid: tx.txid });
+      topic._id = txid;
+      topic.txid = txid;
+      await DBHelper.removeTopic(db.Topics, tx.txid);
```

Comment:
we should remove here. We shouldnt use `txid` as topic._id

Reply:
The txid is already being used for Topic _id since we don't know the address until the Topic is created. The sync logic won't update the Topic if the txid is the approve txid (as opposed to the createEvent txid). And you can't update the _id of an existing item so I opted to replace the entry with a new one. Any suggestions?

Reply:
@dwalintukan yes, just the same way we handle it before, using nedb generated id for `_id`. I will add additional index later

Reply:
Ah I see. I will have to see how to generate the need id. Not familiar with that code. Will change so I update instead of remove the topic and oralce

Reply:
just insert object without `_id` specified

Reply:
:+1: